src/posthog-persistence.ts

```diff
@@ -122,6 +122,9 @@ export class PostHogPersistence {
     }
 
     save(): void {
+        // NOTE: Saving frequently causes issues with Recordings and CMP tools which
+        // observe cookie changes, and modify their UI, often causing infinite loops.
+        // As such callers of this should ideally check that the data has changed beforehand
```

Comment:
```suggestion
/**
* NOTE: Saving frequently causes issues with Recordings and CMP tools which
* observe cookie changes, and modify their UI, often causing infinite loops.
* As such callers of this should ideally check that the data has changed beforehand
*/
    save(): void {
```

like this so people get intellisense?